### PR TITLE
Change board list and board role entity names

### PR DESCRIPTION
**Pull request by ChernykhVyacheslav**
BoardList and BoardRole entity names were too confusing. 
BoardList was changed to BoardColumn.
BoardRole was changed to UserBoardRole

https://trello.com/c/fpC2QISc/162-change-boardlist-and-boardrole-entity-names

### CODE REVIEW

File: application/src/main/java/com/softserve/borda/BordaApplication.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
     UserBoardRelationRepository userBoardRelationRepository;
 
-    BoardRoleRepository boardRoleRepository;
+    UserBoardRoleRepository userBoardRoleRepository;
```
Comment (achehrynets) [inline application/src/main/java/com/softserve/borda/BordaApplication.java:28]: please add incapsulation

File: application/src/main/java/com/softserve/borda/BordaApplication.java [hunk 2]
```diff
@@ -36,22 +36,22 @@
 
     BoardService boardService;
 
-    BoardListService boardListService;
+    BoardColumnService boardColumnService;
```
Comment (achehrynets) [inline application/src/main/java/com/softserve/borda/BordaApplication.java:36]: please add incapsulation

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java [hunk 3]
```diff
@@ -154,11 +154,11 @@ public Board getBoard(@PathVariable Long id) {
     public ResponseEntity<BoardListDTO> createBoardListsForBoard(@PathVariable Long boardId,
                                                                  @RequestBody BoardListDTO boardListDTO) {
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Is it still BoardListDTO?

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java [hunk 4]
```diff
@@ -154,11 +154,11 @@ public Board getBoard(@PathVariable Long id) {
...
         try {
-            BoardList boardList = new BoardList();
-            boardList.setName(boardListDTO.getName());
-            boardList = boardListService.create(boardList);
-            boardList = boardListService.addBoardListToBoard(boardId, boardList.getId());
-            boardListDTO = modelMapper.map(boardList, BoardListDTO.class);
+            BoardColumn boardColumn = new BoardColumn();
+            boardColumn.setName(boardListDTO.getName());
+            boardColumn = boardColumnService.create(boardColumn);
+            boardColumn = boardColumnService.addBoardListToBoard(boardId, boardColumn.getId());
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Does BoardColumn entity (or table) contain reference on Board entity?

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java [hunk 5]
```diff
@@ -269,14 +269,14 @@ public Board getBoard(@PathVariable Long id) {
                                                                      @PathVariable Long ticketId,
                                                                      @PathVariable String boardId) {
         try {
-            BoardList oldBoardList = boardListService.getBoardListById(oldBoardListId);
-            BoardList newBoardList = boardListService.getBoardListById(newBoardListId);
+            BoardColumn oldBoardColumn = boardColumnService.getBoardListById(oldBoardListId);
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: I think `getBoardListById` method is no longer applicable.

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java [hunk 6]
```diff
@@ -269,14 +269,14 @@ public Board getBoard(@PathVariable Long id) {
...
+            BoardColumn newBoardColumn = boardColumnService.getBoardListById(newBoardListId);
             Ticket ticket = ticketService.getTicketById(ticketId);
-            oldBoardList.getTickets().remove(ticket);
-            newBoardList.getTickets().add(ticket);
-            boardListService.update(oldBoardList);
-            newBoardList = boardListService.update(newBoardList);
-            BoardListDTO boardListDTO = modelMapper.map(newBoardList, BoardListDTO.class);
+            oldBoardColumn.getTickets().remove(ticket);
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Better to use JPQL and just update boardId on already existing ticket.

**Review by achehrynets — CHANGES_REQUESTED** [review]
Please, take a look at changes again and be sure that all method, properties and classes are renamed based on requirements.

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 3 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Yes, my bad. Will fix.

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 4 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Not yet. Will be fixed in a separate branch later.

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 5 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Yes, you are right. Will fix

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 6 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Got it, will look into it.

File: application/src/main/java/com/softserve/borda/BordaApplication.java (hunk 1 above)
Comment (ChernykhVyacheslav) [reply application/src/main/java/com/softserve/borda/BordaApplication.java:28]: Data initialization rework has been assigned to separate ticket: https://trello.com/c/TZtFFohN/173-rework-data-initialization

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 4 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Separate ticket has been assigned for this issue: https://trello.com/c/or4iVPK2/152-re-check-entity-mappings-and-fetching

**Review by jules-n — APPROVED** [review]
+1

File: domain/src/main/java/com/softserve/borda/entities/UserBoardRole.java [hunk 7]
```diff
@@ -7,26 +7,26 @@
 
 @Data
 @Entity(name = "board_roles")
```
Comment (achehrynets) [inline domain/src/main/java/com/softserve/borda/entities/UserBoardRole.java]: I think, name of table also needs to be changed.

**Review by achehrynets — APPROVED** [review]
Looks good but I left one question for clarification.
Please, check and decide if it needs to be updated.

**Pull request merged**